### PR TITLE
gtestify test/core/slice

### PR DESCRIPTION
**Pull request by veblush**

### CODE REVIEW

File: test/core/slice/b64_test.cc [hunk 1]
```diff
@@ -120,101 +123,90 @@ static void test_url_safe_unsafe_mismatch_failure(void) {
...
-static void test_unpadded_decode(void) {
+TEST(B64Test, UnpaddedDecode) {
   grpc_slice decoded;
 
   grpc_core::ExecCtx exec_ctx;
   decoded = grpc_base64_decode("Zm9vYmFy", 0);
-  GPR_ASSERT(!GRPC_SLICE_IS_EMPTY(decoded));
-  GPR_ASSERT(grpc_slice_str_cmp(decoded, "foobar") == 0);
+  ASSERT_FALSE(GRPC_SLICE_IS_EMPTY(decoded));
+  ASSERT_EQ(grpc_slice_str_cmp(decoded, "foobar"), 0);
```
Comment (ctiller) [inline test/core/slice/b64_test.cc:175]: `ASSERT_EQ(StringViewFromSlice(decoded), "foobar");`

File: test/core/slice/b64_test.cc [hunk 2]
```diff
@@ -69,47 +72,47 @@ static void test_full_range_encode_decode_b64(int url_safe, int multiline) {
     grpc_core::ExecCtx exec_ctx;
     b64 = grpc_base64_encode(orig, sizeof(orig) - i, url_safe, multiline);
     orig_decoded = grpc_base64_decode(b64, url_safe);
-    GPR_ASSERT(GRPC_SLICE_LENGTH(orig_decoded) == (sizeof(orig) - i));
```
Comment (ctiller) [inline test/core/slice/b64_test.cc:72]: This and the following line could be replaced with:

ASSERT_EQ(StringViewFromSlice(orig_decoded), string_view(orig, sizeof(orig)))

File: test/core/slice/b64_test.cc (hunk 1 above)
Comment (veblush) [reply test/core/slice/b64_test.cc:175]: Done.

File: test/core/slice/b64_test.cc (hunk 2 above)
Comment (veblush) [reply test/core/slice/b64_test.cc:72]: Done.

File: test/core/slice/b64_test.cc [hunk 3]
```diff
@@ -120,101 +120,90 @@ static void test_url_safe_unsafe_mismatch_failure(void) {
...
 
-static void test_unpadded_decode(void) {
+TEST(B64Test, UnpaddedDecode) {
   grpc_slice decoded;
 
   grpc_core::ExecCtx exec_ctx;
   decoded = grpc_base64_decode("Zm9vYmFy", 0);
-  GPR_ASSERT(!GRPC_SLICE_IS_EMPTY(decoded));
-  GPR_ASSERT(grpc_slice_str_cmp(decoded, "foobar") == 0);
+  ASSERT_FALSE(GRPC_SLICE_IS_EMPTY(decoded));
```
Comment (ctiller) [inline test/core/slice/b64_test.cc:171]: this is now a redundant assert (same throughout)

Comment (veblush) [reply test/core/slice/b64_test.cc]: done.

**Review by ctiller — APPROVED** [review]

**Pull request merged**